services/iam/app/controllers/iam/passwords_controller.rb

```diff
@@ -42,8 +42,10 @@ def update
         }
 
         res = User.reset_password_by_token(decoded_params)
+        res.confirm unless res.confirmed? || !res.persisted?
```

Comment:
At this point and since we're already evaluating `if res.persisted?` just below, we can simply update to

```
if res.persisted?
  res.confirm unless res.confirmed?
  @current_jwt ....
else
  ...
```